backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/ShipmentScheduleQtyExternalPropertyStorage.java

```diff
@@ -0,0 +1,122 @@
+import org.adempiere.inout.util.ShipmentScheduleAvailableStockDetail;
+import org.adempiere.warehouse.WarehouseId;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class shall hold details of external property VHUs that are reserved for an OrderLine and ready to be shipped.
+ * These should never show up in {@code MD_Stock} and as such would never be picked up by the qtyonHand logic.
+ */
+public class ShipmentScheduleQtyExternalPropertyStorage implements IShipmentScheduleQtyOnHandStorage
+{
+	private final IUOMConversionBL uomConversionBL = Services.get(IUOMConversionBL.class);
+	private final IProductDAO productDAO = Services.get(IProductDAO.class);
+	private final IHandlingUnitsDAO handlingUnitsDAO = Services.get(IHandlingUnitsDAO.class);
+
+	private final HUReservationRepository huReservationRepository;
+	private final HUDescriptorService huDescriptorService;
+
+	public ShipmentScheduleQtyExternalPropertyStorage(final HUReservationRepository huReservationRepository, final HUDescriptorService huDescriptorService)
+	{
+		this.huReservationRepository = huReservationRepository;
+		this.huDescriptorService = huDescriptorService;
+	}
+
+	@Override
+	@NonNull
+	public List<ShipmentScheduleAvailableStockDetail> getStockDetailsMatching(final @NonNull I_M_ShipmentSchedule sched)
+	{
+		final OrderLineId orderLineId = OrderLineId.ofRepoIdOrNull(sched.getC_OrderLine_ID());
+
+		if (orderLineId == null)
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			// Reserved qty shall always be in VHUs, no need for picking BOM.
+			return huReservationRepository.getByDocumentRef(HUReservationDocRef.ofSalesOrderLineId(orderLineId))
+					.map(huRes -> toShipmentScheduleAvailableStockDetail(huRes, sched))
+					.orElse(ImmutableList.of());
+		}
+	}
+
+	private ImmutableList<ShipmentScheduleAvailableStockDetail> toShipmentScheduleAvailableStockDetail(@NonNull final HUReservation huRes, @NonNull final I_M_ShipmentSchedule sched)
+	{
+		return huRes.getVhuIds()
+				.stream()
+				.filter(this::isExternalPropertyVHU)
+				.map(vhuId -> toShipmentScheduleAvailableStockDetail(vhuId, huRes, sched))
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	private boolean isExternalPropertyVHU(final HuId vhuId)
+	{
+		final ImmutableList<HUDescriptor> huDescriptors = huDescriptorService.createHuDescriptors(vhuId);
+		Check.assumeEquals(huDescriptors.size(), 1, "VHU {} should be associated with exactly 1 M_HU_Storage", vhuId);
+		final HUDescriptor huDescriptor = Iterables.getOnlyElement(huDescriptors);
+		return huDescriptor.isExternalProperty();
+	}
+
+	private ShipmentScheduleAvailableStockDetail toShipmentScheduleAvailableStockDetail(@NonNull final HuId vhuId, final @NonNull HUReservation huRes, @NonNull final I_M_ShipmentSchedule sched)
+	{
+		final ProductId productId = ProductId.ofRepoId(sched.getM_Product_ID());
+		final UomId uomId = UomId.ofRepoId(productDAO.getById(productId).getC_UOM_ID());
```

Comment:
why not using de.metas.product.IProductBL#getStockUOMId(de.metas.product.ProductId) ?

Reply:
u could rename uomId to stockUomId

Reply:
forgot it existed.